shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -86,27 +89,47 @@ data class ExpandedEntity(
             this.plus(propertyKey to JsonLdUtils.buildNonReifiedTemporalValue(dateTime))
         else this
 
-    fun filterOnAttributes(includedAttributes: Set<String>): Map<String, Any> {
+    fun filterOnAttributes(includedAttributes: Set<String>, includedDatasetIds: Set<URI>?): Map<String, Any> {
         val inputToMap = { i: ExpandedEntity -> i.members }
-        return filterEntityOnAttributes(this, inputToMap, includedAttributes)
+        return filterEntityOnAttributes(this, inputToMap, includedAttributes, includedDatasetIds)
     }
 
-    private fun <T> filterEntityOnAttributes(
-        input: T,
-        inputToMap: (T) -> Map<String, Any>,
-        includedAttributes: Set<String>
+    private fun filterEntityOnAttributes(
+
+        input: ExpandedEntity,
+        inputToMap: (ExpandedEntity) -> Map<String, Any>,
+        includedAttributes: Set<String>,
+        includedDatasetIds: Set<URI>?,
     ): Map<String, Any> {
-        return if (includedAttributes.isEmpty()) {
-            inputToMap(input)
-        } else {
+        var result = inputToMap(input)
+
+            if (includedAttributes.isEmpty() && includedDatasetIds == null) {
+                result = inputToMap(input)
```

Comment:
there are some redundant calls to `inputToMap` (here and below). what's more, since this function is now only called by `filterOnAttributes`, it is not even necessary to keep it and you can replace this parameter by `members: Map<String, Any>`.